packages/files-ui/src/Components/Modules/FileBrowsers/CreateSharedFolderModal.tsx

```diff
@@ -0,0 +1,238 @@
+  const [isCreatingSharedFolder, setIsCreatingSharedFolder] = useState(false)
+  const [sharedFolderName, setSharedFolderName] = useState("")
+  const [sharedFolderUsers, setSharedFolderUsers] = useState<Array<{label: string; value: LookupUser}>>([])
+  const [permissions, setPermissions] = useState<"read" | "write" | undefined>(undefined)
+  const { desktop } = useThemeSwitcher()
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (modalOpen) {
+      setTimeout(() => inputRef.current?.focus(), 100)
+    }
+  }, [modalOpen])
+
+  const handleLookupUser = useCallback(async (inputVal: string) => {
+    if (inputVal === "") return []
+    const lookupBody: LookupUserRequest = {}
+    const ethAddressRegex = new RegExp("^0(x|X)[a-fA-F0-9]{40}$") // Eth Address Starting with 0x and 40 HEX chars
+    const pubKeyRegex = new RegExp("^0(x|X)[a-fA-F0-9]{66}$") // Compressed public key, 66 chars long
+
+    if (ethAddressRegex.test(inputVal)) {
+      lookupBody.public_address = inputVal
+    } else if (pubKeyRegex.test(inputVal)) {
+      lookupBody.identity_public_key = inputVal
+    } else {
+      lookupBody.username = inputVal
+    }
+
+    const result = await filesApiClient.lookupUser(lookupBody)
+
+    if (!result) return []
+    const currentUsers = Array.isArray(sharedFolderUsers) ? sharedFolderUsers.map(su => su.value.uuid) : []
+    if (currentUsers.includes(result.uuid)) return []
+
+    return [{ label: inputVal, value: result }]
+  }, [filesApiClient, sharedFolderUsers])
+
+  const handleCreateSharedFolder = useCallback(async () => {
+    const users = sharedFolderUsers.map(su => ({
+      uuid: su.value.uuid,
+      pubKey: EthCrypto.publicKey.decompress(su.value.identity_pubkey.slice(2))
+    }))
+    const readers = (permissions === "read") ? users : []
+    const writers = (permissions === "write") ? users : []
+    setIsCreatingSharedFolder(true)
+    createSharedFolder(sharedFolderName, writers, readers)
+      .then(close)
+      .catch(console.error)
+      .finally(() => setIsCreatingSharedFolder(false))
+  }, [sharedFolderUsers, createSharedFolder, permissions, sharedFolderName, close])
+
+  return (
+    <CustomModal
+      className={classes.modalRoot}
+      injectedClass={{
+        inner: classes.modalInner
+      }}
+      active={modalOpen}
+      closePosition="none"
+      maxWidth="sm"
+    >
+      <div className={classes.root}>
+        <div className={classes.iconBacking}>
+          <ShareAltSvg />
+        </div>
+        <div className={classes.heading}>
+          <Typography variant='body1'>
+            <Trans>Create Shared Folder</Trans>
+          </Typography>
+        </div>
+        <div className={classes.modalFlexItem}>
+          <TextInput
+            ref={inputRef}
+            label={t`Shared Folder Name`}
+            value={sharedFolderName}
+            onChange={(value) => {value && setSharedFolderName(value.toString())}} />
```

Comment:
prevents the first letter not going away issue!
```suggestion
            onChange={(value) => {setSharedFolderName(value?.toString() || "")}} />
```